### PR TITLE
Resonator fields now visually show how long they have until they burst

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
tweak: Resonator fields now visually show how long they have until they burst.
bugfix: Hitting a legion skull with a resonator will now produce a field.
/:cl:

Hmm, mmm, maybe part of why it's so bad is that you can't tell when it's going to burst?

### CODE REVIEW

**Comment by adamkad1** [comment]
still will suck on legions

**Comment by Repukan** [comment]
Hugbox

**Comment by adamkad1** [comment]
mayby give plasma cutters some love? one (probally) bug made me fire both plasma cutters i held at once, although i couldnt hit broad side of barn if not in point blank, it should totally be a mechanic. otherwise mayby some mods for them.

**Comment by Repukan** [comment]
Plasma cutters are fucking op and need a nerf. It's not a bug, you probably had harm intent on and dual wielding kicked in.

**Comment by Kromgar** [comment]
@Repukan Shut your whore mouth

**Comment by Repukan** [comment]
Fucking fight me you plebe bitch

File: code/modules/mining/equipment.dm [hunk 1]
```diff
@@ -208,64 +196,93 @@
...
-		if(!check_allowed_items(target, 1))
-			return
+/obj/item/weapon/resonator/proc/CreateResonance(target, mob/user)
+	var/turf/T = get_turf(target)
+	var/obj/effect/temp_visual/resonance/R = locate(/obj/effect/temp_visual/resonance) in T
+	if(R)
+		R.damage_multiplier = quick_burst_mod
+		R.burst()
+		return
+	if(LAZYLEN(fields) < fieldlimit)
```
Comment (Cyberboss) [inline code/modules/mining/equipment.dm]: Fields always exists. Use .len

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mining/equipment.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/modules/mining/equipment.dm]: I thought this was faster anyway?

**Comment by adamkad1** [comment]
hey krom, why be a asshole?

**Pull request merged**